### ISSUE TITLE
Add simple_efficiency tests to cover numpy arrays and pandas Series

### ISSUE THREAD

**Issue by aman-coder03** [pvlib/pvlib-python#2649]
### Description

`pvlib.transformer.simple_efficiency` supports vectorized operations via
NumPy, but the current test coverage primarily exercises scalar inputs.

This issue proposes adding tests that explicitly cover
`numpy.ndarray` inputs for:

- `input_power`
- `load_loss`
- `no_load_loss`

including the edge case `load_loss = 0`.

**Comment by echedey-ls** [comment]
For reference, this is the file that contains the tests: https://github.com/pvlib/pvlib-python/blob/main/tests/test_transformer.py
It is indeed testing with `pd.Series` and scalar inputs. I don't oppose nor support checking extra types for simplicity.

**Comment by cwhanse** [comment]
@echedey-ls I think the three device parameters are only tested as `float`, not for the range of `numeric` described in the docstring.

**Comment by echedey-ls** [comment]
@cwhanse you are right. At first I didn't think it would make much sense to run this function on many transformers at the same time, but maybe it does to pick the best one from a list to fit an specific scenario. I still wonder if added complexity is beneficial. I think documenting this broadcast via a small docstring example is more important than testing it.

**Comment by cwhanse** [comment]
Many transformers, or perhaps the loss parameters vary with time (temperature dependent?). Either way, it's not the usual use of the function, and we could just change the docstring type from `numeric` to `float`.

**Comment by echedey-ls** [comment]
> Many transformers, or perhaps the loss parameters vary with time (temperature dependent?).

AFAIK, loss parameters are just given at nominal conditions. While physically there surely is a dependence on temperature, I hardly think that is measured frequently and with enough reproducibility. Lab testing according to IEC 60076-1 does not include that.

> we could just change the docstring type from numeric to float.

I also support this.

**Comment by markcampanelli** [comment]
I note that not supporting vectorized parameters could make the function harder to use for sensitivity analyses and uncertainty quantification.

**Comment by cwhanse** [comment]
@markcampanelli I was thinking that the length of parameter vectors for `load_loss` and `no_load_loss` would match the length of `input_power`, i.e., the parameters are indexed by time rather than representing alternative values.

**Comment by mikofski** [comment]
In my experience the load/no-load parameters are constant.

**Comment by kumaradityaapril** [comment]
I appreciate the conversation; this background is useful.

It seems reasonable to refrain from suggesting complete vectorization support for the loss parameters based on the comments thus far. Adding tests that cover `input_power` as a NumPy array (reflecting time-varying load) while maintaining `load_loss` and `no_load_loss` as scalars could be a compromise.
Without enlarging the intended API, this would increase coverage.

If that strategy seems appropriate, I'd be happy to work on it.

**Comment by cwhanse** [comment]
@kumaradityaapril that would be welcome! Scope is to increase test coverage and change the docstring descriptions for `load_loss` and `no_load_loss` from numeric to float.

**Comment by kumaradityaapril** [comment]
I appreciate the explanation. I'm excited to work on this.
As advised, I'll concentrate on expanding test coverage and updating the docstrings for load_loss and no_load_loss.
